File: src/myjava/ReverseString.java
package myjava;

public class ReverseString{

    public static void main(String[] args) {
        String s = "apple";
        System.out.println(s);
        System.out.println(reverse(s));

    }

    public static String reverse(String s) {
        if (s == null || s.length() - 1 == 0) return s;
        return s.charAt(s.length() - 1) + reverse(s.substring(0, s.length() - 1));
    }

}

File: src/mykotlin/ReverseString.kt
package mykotlin

fun main() {
    println(reverseString("apple"))
}

fun reverseString(s: String): String {
    if(s.length <= 1) return s
    return reverseString(s.substring(1, s.length)) + s[0]
}


// Nullable version of reverse string
fun reverseStringNullable(s: String?): String? {
    if(s.isNullOrBlank()) return null
    else {
        if(s.length <= 1) return s
        return reverseString(s.substring(1, s.length)) + s[0]
    }
}